delta-lake/src/main/java/org/apache/iceberg/delta/utils/DeltaLakeDataTypeVisitor.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.delta.utils;
+
+import io.delta.standalone.types.ArrayType;
+import io.delta.standalone.types.DataType;
+import io.delta.standalone.types.MapType;
+import io.delta.standalone.types.StructField;
+import io.delta.standalone.types.StructType;
+import java.util.List;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+
+public abstract class DeltaLakeDataTypeVisitor<T> {
```

Comment:
can this just be protected so we don't make a new public class? (same to DeltaLakeTypeToType). We probably need to move this to the parent package to make it work, which is fine